core/string/ustring.cpp

```diff
@@ -3986,54 +3986,161 @@ String String::format(const Variant &values, const String &placeholder) const {
 	return new_string;
 }
 
-String String::replace(const String &p_key, const String &p_with) const {
-	String new_string;
+static String _replace_common(const String &p_this, const String &p_key, const String &p_with, bool p_case_insensitive) {
```

Comment:
Since the code only differs on which `find` method to use I created a helper method to avoid code duplication